### ISSUE TITLE
bug: the release assets file structure is changed between 0.14.1-rc.1 and 0.14.1-rc.2 [regression]

### ISSUE THREAD

**Issue by hydai** [WasmEdge/WasmEdge#3684]
### Summary

There are regressions between 0.14.1-rc.1 and 0.14.1-rc.2

![CleanShot 2024-08-22 at 14 45 16](https://github.com/user-attachments/assets/1fe01f37-b231-48ca-ab69-a1c5aa0d29f4)
![CleanShot 2024-08-22 at 14 47 03](https://github.com/user-attachments/assets/407b305c-42ae-4d25-8766-6bbbaebce0c4)

### Current State

0.14.1-rc.1 macOS
```
WasmEdge-0.14.1-rc.1-Darwin
├── bin
│   ├── wasmedge
│   └── wasmedgec
├── include
│   └── wasmedge
│       ├── enum.inc
│       ├── enum_configure.h
│       ├── enum_errcode.h
│       ├── enum_types.h
│       ├── int128.h
│       ├── version.h
│       └── wasmedge.h
└── lib
    ├── libwasmedge.0.1.0.dylib
    ├── libwasmedge.0.1.0.tbd
    ├── libwasmedge.0.dylib -> libwasmedge.0.1.0.dylib
    ├── libwasmedge.0.tbd -> libwasmedge.0.1.0.tbd
    ├── libwasmedge.dylib -> libwasmedge.0.dylib
    └── libwasmedge.tbd -> libwasmedge.0.tbd
```

0.14.1-rc.2 macOS
```
WasmEdge-0.14.1-rc.2-darwin_arm64
├── bin
│   ├── wasmedge
│   └── wasmedgec
├── include
│   └── wasmedge
│       ├── enum.inc
│       ├── enum_configure.h
│       ├── enum_errcode.h
│       ├── enum_types.h
│       ├── int128.h
│       ├── version.h
│       └── wasmedge.h
└── lib
    ├── libwasmedge.0.1.0.dylib
    ├── libwasmedge.0.1.0.tbd
    ├── libwasmedge.0.dylib -> libwasmedge.0.1.0.dylib
    ├── libwasmedge.0.tbd -> libwasmedge.0.1.0.tbd
    ├── libwasmedge.dylib -> libwasmedge.0.dylib
    └── libwasmedge.tbd -> libwasmedge.0.tbd
```

### Expected State

The folder name should be `WasmEdge-<Version>-Darwin` and `WasmEdge-<Version>-Linux`. Otherwise, the installer will fail.

### Reproduction steps

1. Download assets from the 0.14.1-rc.1 and 0.14.1-rc.2
2. Unzip these assets, and you can find the result

### Screenshots

![DESCRIPTION](LINK.png)

### Components

CLI, C SDK

### WasmEdge Version or Commit you used

0.14.1-rc.2

### Operating system information

Linux and macOS

### Hardware Architecture

All architecture

**Comment by q82419** [comment]
Maybe broken when apply the cpack component.
I'll check and try to fix.

**Comment by q82419** [comment]
Comment:

The reason is that in `rc.1`, the files are under the `WasmEdge-0.14.1-rc.1-Darwin` folder:

```
$ tar -tvf WasmEdge-0.14.1-rc.1-darwin_arm64.tar.gz
drwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/bin/
-rwxr-xr-x  0 runner staff   33480  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/bin/wasmedge
-rwxr-xr-x  0 runner staff   33480  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/bin/wasmedgec
drwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/include/
drwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/
-rw-r--r--  0 runner staff  172341  8  9 02:00 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/wasmedge.h
-rw-r--r--  0 runner staff     773  8  9 02:01 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/version.h
-rw-r--r--  0 runner staff   46258  8  9 02:01 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/enum.inc
-rw-r--r--  0 runner staff    2109  8  9 02:01 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/enum_configure.h
-rw-r--r--  0 runner staff     976  8  9 02:01 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/enum_errcode.h
-rw-r--r--  0 runner staff     912  8  9 02:00 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/int128.h
-rw-r--r--  0 runner staff    1208  8  9 02:01 WasmEdge-0.14.1-rc.1-Darwin/include/wasmedge/enum_types.h
drwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/
lrwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/libwasmedge.0.dylib -> libwasmedge.0.1.0.dylib
lrwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/libwasmedge.tbd -> libwasmedge.0.tbd
lrwxr-xr-x  0 runner staff       0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/libwasmedge.dylib -> libwasmedge.0.dylib
-rw-r--r--  0 runner staff 4047354  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/libwasmedge.0.1.0.tbd
-rwxr-xr-x  0 runner staff 160461800  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/libwasmedge.0.1.0.dylib
lrwxr-xr-x  0 runner staff         0  8  9 02:03 WasmEdge-0.14.1-rc.1-Darwin/lib/libwasmedge.0.tbd -> libwasmedge.0.1.0.tbd
```

But after applying the package component, the folder disappeared:

```
$ tar -tvf ~/Downloads/WasmEdge-0.14.1-rc.2-darwin_arm64.tar.gz
drwxr-xr-x  0 runner staff       0  8 20 23:08 bin/
-rwxr-xr-x  0 runner staff   33464  8 20 23:08 bin/wasmedge
-rwxr-xr-x  0 runner staff   33464  8 20 23:08 bin/wasmedgec
drwxr-xr-x  0 runner staff       0  8 20 23:08 include/
drwxr-xr-x  0 runner staff       0  8 20 23:08 include/wasmedge/
-rw-r--r--  0 runner staff   46258  8 20 23:06 include/wasmedge/enum.inc
-rw-r--r--  0 runner staff    2109  8 20 23:06 include/wasmedge/enum_configure.h
-rw-r--r--  0 runner staff     976  8 20 23:06 include/wasmedge/enum_errcode.h
-rw-r--r--  0 runner staff    1208  8 20 23:06 include/wasmedge/enum_types.h
-rw-r--r--  0 runner staff     912  8 20 23:06 include/wasmedge/int128.h
-rw-r--r--  0 runner staff     773  8 20 23:06 include/wasmedge/version.h
-rw-r--r--  0 runner staff  172341  8 20 23:06 include/wasmedge/wasmedge.h
drwxr-xr-x  0 runner staff       0  8 20 23:08 lib/
-rwxr-xr-x  0 runner staff 160145144  8 20 23:08 lib/libwasmedge.0.1.0.dylib
-rw-r--r--  0 runner staff   3943152  8 20 23:08 lib/libwasmedge.0.1.0.tbd
lrwxr-xr-x  0 runner staff         0  8 20 23:08 lib/libwasmedge.0.dylib -> libwasmedge.0.1.0.dylib
lrwxr-xr-x  0 runner staff         0  8 20 23:08 lib/libwasmedge.0.tbd -> libwasmedge.0.1.0.tbd
lrwxr-xr-x  0 runner staff         0  8 20 23:08 lib/libwasmedge.dylib -> libwasmedge.0.dylib
lrwxr-xr-x  0 runner staff         0  8 20 23:08 lib/libwasmedge.tbd -> libwasmedge.0.tbd
```

**Comment by q82419** [comment]
Revert the cpack changes in `0.14.x` versions: #3689

And the installer and cpack will be updated in `0.15.0`: #3690.

**Comment by hydai** [comment]
This is fixed by #3689
